### DISCUSSION TITLE
Using method in select in a linq query should still work in this way?

### DISCUSSION THREAD

**Discussion by beercsaba** [linq2db/linq2db#4465] [General]
Hi,

I would like to write linq queries in this way and 'materialize the objects' by using a method call after `select`:
```
      public IQueryable<ViewRow> OperationLogView(CultureInfo cultureInfo) =>
         from opl in dbCtx.Log.OperationLogs
         from us in dbCtx.Human.UserData.InnerJoin(j => j.Id == opl.InitiatorUserId)
         select ViewRowBuilderFunc(opl, us, cultureInfo)
```
other way can be the constructor call in some way like this:
```select new ViewRow(opl.col1, opl.col2, ...) { ComputedCol= us.col4 + opl.col11}```

i would rather use a method after `select` instead of using a new statement, because it is more elegant and reuseable for me.

I've found this solution [here](https://stackoverflow.com/questions/63727166/linq2db-filter-by-nested-property-field) from 2020, and followed the way described in the answer, but unfortunately this does not work for me with the latest version of linq2db (5.4.0). I always get this NRE in QueryRunner class:
![image](https://github.com/linq2db/linq2db/assets/24902130/56be5556-41c3-4170-ad28-607b9bb351bb)

Should this kind of 'materialization' work with latest linq2db versions too or some breaking change has happened in the past 4 years?

Thanks, best regards,
Csaba

**Comment by MaceWindu** [comment]
This type of mapping is supported. You should check in debugger where is null reference comes from and if it is linq2db issue - fill new issue

**Comment by beercsaba** [comment]
Hi @MaceWindu,

Thanks for your reply.
